scripts/gitlab/build_and_test.sh

```diff
@@ -112,7 +112,7 @@ then
     $cmake_exe \
       -C ${hostconfig_path} \
       ${project_dir}
-    if ! $cmake_exe --build . -j; then
+    if ! $cmake_exe --build . -j 12 ; then
```

Comment:
Why the change to `-j 12`?

Reply:
I think @kab163 changed the number of jobs for RAJA (from `-j` to `-j32` or so). I thought I would do the same here. Although not related to this PR...

Reply:
I can revert that for sure.

Reply:
Done